decidim/decidim
# Fix admin password change required for omniauth-only accounts

#### :tophat: What? Why?
When some admins are only using the omniauth logins, they are still asked to perform a password change after their password expires. The problem is further explained at #10398.

There are also other issues related to the admin password changes that I am fixing at #10492. That PR has to be incorporated with this PR for the fix to actually work for the described issue at #10398.

#### :pushpin: Related Issues
- Related to #9347, #10492
- Fixes #10398

#### Testing
- Configure an omniauth authentication method to your testing/development instance
- Make sure the user you are about to authenticate with does not yet exist in Decidim
- Authenticate through omniauth using that account
- See that the system is no longer asking you to change your password straight after the authentication
- Log out
- From rails console, update that user's `password_updated_at` e.g. to `1000.years.ago`
- Authenticate again through omniauth using that account
- See that the system is now asking you to change your password straight after the authentication (assuming the user has manually added a password to their account in which case we should regularly ask the user to update their password)

In case you go ahead and update the password manually from the account update section, you will have to regularly change that password.
